modules/greenbidsBidAdapter.js

```diff
@@ -0,0 +1,269 @@
+  let ref = '';
+  if (bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.page) {
+    ref = bidderRequest.refererInfo.page;
+  }
+  return ref;
+}
+
+function getPageTitle() {
+  try {
+    const ogTitle = window.top.document.querySelector('meta[property="og:title"]');
+    return window.top.document.title || (ogTitle && ogTitle.content) || '';
+  } catch (e) {
+    const ogTitle = document.querySelector('meta[property="og:title"]');
+    return document.title || (ogTitle && ogTitle.content) || '';
+  }
+}
+
+function getPageDescription() {
+  try {
+    const element = window.top.document.querySelector('meta[name="description"]') ||
+      window.top.document.querySelector('meta[property="og:description"]');
+    return (element && element.content) || '';
+  } catch (e) {
+    const element = document.querySelector('meta[name="description"]') ||
+      document.querySelector('meta[property="og:description"]');
+    return (element && element.content) || '';
+  }
+}
+
+function getConnectionDownLink(nav) {
+  return nav && nav.connection && nav.connection.downlink >= 0 ? nav.connection.downlink.toString() : '';
+}
+
+function getTimeToFirstByte(win) {
+  const performance = win.performance || win.webkitPerformance || win.msPerformance || win.mozPerformance;
+
+  const ttfbWithTimingV2 = performance &&
+    typeof performance.getEntriesByType === 'function' &&
+    Object.prototype.toString.call(performance.getEntriesByType) === '[object Function]' &&
+    performance.getEntriesByType('navigation')[0] &&
+    performance.getEntriesByType('navigation')[0].responseStart &&
+    performance.getEntriesByType('navigation')[0].requestStart &&
+    performance.getEntriesByType('navigation')[0].responseStart > 0 &&
+    performance.getEntriesByType('navigation')[0].requestStart > 0 &&
+    Math.round(
+      performance.getEntriesByType('navigation')[0].responseStart - performance.getEntriesByType('navigation')[0].requestStart
+    );
+
+  if (ttfbWithTimingV2) {
+    return ttfbWithTimingV2.toString();
+  }
+
+  const ttfbWithTimingV1 = performance &&
+    performance.timing.responseStart &&
+    performance.timing.requestStart &&
+    performance.timing.responseStart > 0 &&
+    performance.timing.requestStart > 0 &&
+    performance.timing.responseStart - performance.timing.requestStart;
+
+  return ttfbWithTimingV1 ? ttfbWithTimingV1.toString() : '';
+}
+
+function cleanBidsInfo(bids) {
+  const reqObj = {};
+  let placementId = getValue(bids.params, 'placementId');
+  const gpid = deepAccess(bids, 'ortb2Imp.ext.gpid');
+  reqObj.sizes = getSizes(bids);
+  reqObj.bidId = getBidIdParameter('bidId', bids);
+  reqObj.bidderRequestId = getBidIdParameter('bidderRequestId', bids);
+  reqObj.placementId = parseInt(placementId, 10);
+  reqObj.adUnitCode = getBidIdParameter('adUnitCode', bids);
+  reqObj.transactionId = bids.ortb2Imp?.ext?.tid || '';
+  if (gpid) { reqObj.gpid = gpid; }
+  return reqObj;
```

Comment:
docs: what do you want to "clean" ? What is the input and the output ?
This method is actually only used once. Does it make sense to extract it. I'm not used to Javascript module system, but here you "publicly" expose a method that should be documented.